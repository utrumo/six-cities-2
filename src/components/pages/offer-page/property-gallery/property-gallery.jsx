import React from 'react';
import PropTypes from 'prop-types';

import {connect} from 'react-redux';
import {getCurrentOfferImagesForGallery} from 'store/data/selectors.js';

const PropertyGallery = ({images}) => (
  <div className="property__gallery-container container">
    <div className="property__gallery">
      {images.map((src, i) => (
        <div key={`${i}-${src}`} className="property__image-wrapper">
          <img className="property__image" src={src} alt="Photo studio" />
        </div>
      ))}
    </div>
  </div>
);

PropertyGallery.propTypes = {
  images: PropTypes.arrayOf(PropTypes.string).isRequired,
};

const mapStateToProps = (state) => ({
  images: getCurrentOfferImagesForGallery(state),
});

export {PropertyGallery};
export default connect(mapStateToProps)(PropertyGallery);
