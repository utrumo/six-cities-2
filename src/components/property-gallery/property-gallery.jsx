import React from 'react';
import PropTypes from 'prop-types';

import {connect} from 'react-redux';
import Selectors from '../../store/selectors.js';

import {ASSETS_PATCH} from '../../shared/const';


const PropertyGallery = ({images}) => (
  <div className="property__gallery-container container">
    <div className="property__gallery">
      {images.map((src, i) => (
        <div key={i} className="property__image-wrapper">
          <img className="property__image" src={`${ASSETS_PATCH}${src}`} alt="Photo studio" />
        </div>
      ))}
    </div>
  </div>
);

PropertyGallery.propTypes = {
  images: PropTypes.arrayOf(PropTypes.string).isRequired
};

const mapStateToProps = (state) => ({
  images: Selectors.getCurrentOfferImagesForGallery(state)
});

export {PropertyGallery};
export default connect(mapStateToProps)(PropertyGallery);
