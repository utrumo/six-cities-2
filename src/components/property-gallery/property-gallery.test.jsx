import React from 'react';
import renderer from 'react-test-renderer';
import {PropertyGallery} from './property-gallery.jsx';

it(`PropertyGallery renders correctly`, () => {
  const images = [
    `/img/apartment-02.jpg`,
    `/img/room.jpg`,
    `/img/apartment-01.jpg`,
    `/img/studio-01.jpg`,
    `/img/apartment-03.jpg`,
    `/img/apartment-01.jpg`,
    `/img/apartment-05.jpg`
  ];
  const tree = renderer.create(<PropertyGallery images={images} />) .toJSON();
  expect(tree).toMatchSnapshot();
});
