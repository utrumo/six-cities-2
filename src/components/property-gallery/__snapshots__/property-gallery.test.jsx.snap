// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyGallery renders correctly 1`] = `
<div
  className="property__gallery-container container"
>
  <div
    className="property__gallery"
  >
    <div
      className="property__image-wrapper"
    >
      <img
        alt="Photo studio"
        className="property__image"
        src="/img/apartment-02.jpg"
      />
    </div>
    <div
      className="property__image-wrapper"
    >
      <img
        alt="Photo studio"
        className="property__image"
        src="/img/room.jpg"
      />
    </div>
    <div
      className="property__image-wrapper"
    >
      <img
        alt="Photo studio"
        className="property__image"
        src="/img/apartment-01.jpg"
      />
    </div>
    <div
      className="property__image-wrapper"
    >
      <img
        alt="Photo studio"
        className="property__image"
        src="/img/studio-01.jpg"
      />
    </div>
  </div>
</div>
`;
